src/Microsoft.ML.PCA/PcaTransform.cs

```diff
@@ -190,90 +281,104 @@ private static VersionInfo GetVersionInfo()
         {
             return new VersionInfo(
                 modelSignature: "PCA FUNC",
-                verWrittenCur: 0x00010001, // Initial
+                //verWrittenCur: 0x00010001, // Initial
+                verWrittenCur: 0x00010002, // Get rid of writing float size in model context
                 verReadableCur: 0x00010001,
                 verWeCanReadBack: 0x00010001,
                 loaderSignature: LoaderSignature,
                 loaderAssemblyName: typeof(PcaTransform).Assembly.FullName);
         }
 
-        // These are parallel to Infos.
-        private readonly ColumnType[] _types;
+        private readonly int _numColumns;
+        private readonly ColumnInfo[] _columns;
         private readonly TransformInfo[] _transformInfos;
 
-        private readonly int[] _oversampling;
-        private readonly bool[] _center;
-        private readonly int[] _weightColumnIndex;
-
         private const string RegistrationName = "Pca";
 
         /// <summary>
         /// Public constructor corresponding to SignatureDataTransform.
```

Comment:
>corresponding to SignatureDataTransform [](start = 31, length = 39)

this is no longer true.
Also, since PCA is trainable, it should not have any public constructor: the only way to instantiate it should be via `Fit` of a corresponding estimator.

Reply:
changed to internal

---
In reply to: [227069919](https://github.com/dotnet/machinelearning/pull/1333#discussion_r227069919) [](ancestors = 227069919)